src/components/crossDevice/MobileFlow.js

```diff
@@ -39,9 +42,14 @@ class MobileFlow extends Component {
     this.sendMessage('disconnect pong', data.roomId)
   }
 
-  render = (props) =>
-    this.props.clientSuccess ?
-      <CrossDeviceSubmit {...props}/> : <MobileConnected {...props}/>
+  render = (props) => {
+    if (this.props.clientSuccess)
+      return <CrossDeviceSubmit {...props}/>
+    if (this.props.mobileConnected)
+      return <MobileConnected {...props}/>
+    if (!this.props.mobileConnected)
```

Comment:
Why have you got this as 

```
if(x)
  return ...
if(!x)
  return ...
```

rather than

```
if(x)
  return ...
else
  return ...
```

or even

```
if(x)
  return ...

return ...
```

Sort of nitpicking, but sort of not - since when I first looked at this I assumed you had a render case not covered, until I saw that all if blocks necessarily had to be hit

Reply:
Yes, I did, but then the value changed and I forgot to change it to an if else statement